client/signup/steps/difm-site-picker/site-delete-confirmation-dialog.tsx

```diff
@@ -0,0 +1,165 @@
+		></path>
+		<path
+			stroke="#003C65"
+			strokeLinecap="round"
+			strokeLinejoin="round"
+			strokeMiterlimit="10"
+			strokeWidth="1.8"
+			d="M88.884 44.31c-.986 57.753-86.898 57.753-87.884 0 1.002-57.755 86.914-57.738 87.884 0z"
+		></path>
+	</svg>
+);
+
+const Row = styled.div`
+	display: flex;
+	justify-content: center;
+`;
+
+const DialogContainer = styled.div`
+	text-align: left;
+	width: 600px;
+	padding: 65px 65px 0 65px;
+`;
+
+const DialogButton = styled( Button )`
+	border-radius: 5px;
+	width: 152px;
+	--color-accent: #117ac9;
+	--color-accent-60: #0e64a5;
+	.gridicon {
+		margin-left: 10px;
+	}
+`;
+const AnchorButton = styled( Button )`
+	background: none;
+	border: none;
+	color: var( --color-primary-light );
+	padding: 0;
+`;
+const Subtitle = styled.p`
+	color: var( --color-neutral-50 );
+	font-size: 0.875rem;
+`;
+
+function SiteDeleteConfirmationDialog( {
+	onClose,
+	onConfirm,
+	siteId,
+	onRedirectToNewSite,
+}: {
+	onClose: () => void;
+	onConfirm: () => void;
+	onRedirectToNewSite: () => void;
+	siteId: number;
+} ) {
+	const [ confirmText, setConfirmText ] = useState( '' );
+	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ?? 0 ) );
+	const siteTitle = useSelector( ( state ) => getSiteTitle( state, siteId ) );
+
+	const translate = useTranslate();
+	const deleteDisabled = confirmText !== 'DELETE';
+
+	return (
+		<Dialog
+			isBackdropVisible={ true }
+			isVisible={ true }
+			buttons={ [
+				<DialogButton onClick={ onClose }>{ translate( 'Cancel' ) }</DialogButton>,
+				<DialogButton primary disabled={ deleteDisabled } onClick={ onConfirm }>
+					{ translate( 'Delete site content' ) }
+				</DialogButton>,
+			] }
+			onClose={ onClose }
+		>
+			<DialogContainer>
+				<Row>
+					<ExclamationIcon />
+				</Row>
+				<FormattedHeader
+					align="center"
+					brandFont
+					headerText={ translate( 'Site Reset Confirmation' ) }
+				/>
+				<Subtitle>
+					<ul>
+						<li>
+							{ translate(
+								'The current content of your website {{strong}}%(siteTitle)s{{/strong}} (%(siteAddress)s) will be deleted. ' +
+									'This includes pages, posts, media, comments, third party plugins and themes.',
+								{
+									components: {
+										strong: <strong />,
+									},
+									args: {
+										siteTitle,
+										siteAddress: siteDomain,
+									},
+								}
+							) }
+						</li>
+						<li>
+							{ translate(
+								'Any WordPress.com subscriptions on your site such as custom domains and emails will not be affected.',
+								{
+									components: {
+										strong: <strong />,
+									},
+									args: {
+										siteTitle,
+										siteAddress: siteDomain,
+									},
+								}
+							) }
+						</li>
+						<li>
+							{ translate(
+								"If you don't want your site content to be deleted you can create a {{a}}new site{{/a}} instead.",
+								{
+									components: {
+										a: <AnchorButton onClick={ onRedirectToNewSite } />,
```

Comment:
Instead of `onClick`, maybe we can directly add a `href` attribute to `/start/do-it-for-me`?